### PR TITLE
Feat: syncing page

### PR DESCRIPTION
**Pull request by sorodrigo**
This PR includes the following:
- [x] New Sync modal page component
- [x] Simplifies Home logic
- [x] Addresses the logout racing condition
- [x] Makes `getUser`, `getReportQuestions`, `getFeedbackQuestions`, `getCountries` offline first!

### CODE REVIEW

File: app/containers/sync/index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+function mapStateToProps(state) {
+  return {
+    isConnected: state.offline.online,
+    reach: state.app.netInfo.reach,
+    languageChanged: state.app.language !== getLanguage(),
+    hasAreas: state.areas.data && state.areas.data.length > 0 && state.areas.synced,
+    hasUserData: state.user.data && Object.keys(state.user.data).length > 0 && state.user.synced,
+    hasForms: {
+      report: Object.keys(state.reports.forms).length > 0 && state.reports.synced,
```
Comment (j8seangel) [inline app/containers/sync/index.js]: I think we can set the sync flag as false by default and remove the forms.length checks to make it easier to understand and avoid crashes if the forms data is not an object so I would change the name as well calling it `syncPending` or something like that.

File: app/containers/sync/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      weekly: () => dispatch(getFeedbackQuestions('weekly'))
+    }
+  };
+}
+
+function mergeProps({ hasUserData, hasAreas, hasForms, languageChanged, ...stateProps },
+  { getUserDispatch, getAreasDispatch, getFormsDispatch, ...dispatchProps }, ownProps) {
+  return {
+    ...ownProps,
+    ...dispatchProps,
```
Comment (j8seangel) [inline app/containers/sync/index.js]: Do we need this?

File: app/containers/sync/index.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    ...stateProps,
+    getUser: () => (!hasUserData && getUserDispatch()),
```
Comment (j8seangel) [inline app/containers/sync/index.js]: Do you like the idea to move this to a new action so we will check it there instead to check all of the flags in the component? Also if we move the logic to an action we will be able to sync the app from any place.

File: app/redux-modules/feedback.js [hunk 4]
```diff
@@ -14,14 +15,25 @@ const UPDATE_FEEDBACK = 'feedback/UPDATE_FEEDBACK';
...
+        form = action.payload.data[0].attributes;
+      }
+      if (form && form.questions && form.questions.length) {
+        form.questions = form.questions.sort((a, b) => parseInt(a.order, 10) - parseInt(b.order, 10));
+      }
+      const synced = `${action.meta.type}Synced`;
       const feedback = {
-        [action.payload.type]: action.payload.data
+        [action.meta.type]: form,
+        [synced]: true
```
Comment (j8seangel) [inline app/redux-modules/feedback.js]: Loving this syntax

File: app/redux-modules/feedback.js [hunk 5]
```diff
@@ -14,14 +15,25 @@ const UPDATE_FEEDBACK = 'feedback/UPDATE_FEEDBACK';
...
-  list: {}
+  list: {},
+  dailySynced: false,
+  weeklySynced: false
 };
 
 export default function reducer(state = initialNavState, action) {
   switch (action.type) {
-    case GET_QUESTIONS: {
+    case GET_FEEDBACK_QUESTIONS_COMMIT: {
```
Comment (j8seangel) [inline app/redux-modules/feedback.js]: We need to add a case for the reducer in the `GET_FEEDBACK_QUESTIONS_REQUEST` to set the synced flag as false.

**Comment by sorodrigo** [comment]
🤙🤙🤙

**Comment by j8seangel** [comment]
Awesome work!!!

**Pull request merged**